nautobot/ipam/tests/test_filters.py

```diff
@@ -837,12 +834,14 @@ def setUpTestData(cls):
         )
 
     def test_name(self):
-        params = {"name": VLAN.objects.all().values_list("name", flat=True)[:2]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
+        names = list(VLAN.objects.all().values_list("name", flat=True))[:2]
+        params = {"name": names}
+        self.assertQuerysetEqual(self.filterset(params, self.queryset).qs, self.queryset.filter(name__in=names))
 
     def test_vid(self):
-        params = {"vid": VLAN.objects.all().values_list("vid", flat=True)[:3]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 3)
+        vids = list(VLAN.objects.all().values_list("vid", flat=True))[:3]
```

Comment:
Without casting to a list, `values_list` returns a QuerySet. This is fine for the `self.filterset` call, but on MySQL the `self.queryset.filter()` call chokes if given a sliced QuerySet (rather than a list), e.g.:

```python
>>> VLAN.objects.filter(vid__in=VLAN.objects.all().values_list("vid", flat=True)[:3])
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    return self.cursor.execute(query, args)
  File "/usr/local/lib/python3.7/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.7/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/usr/local/lib/python3.7/site-packages/MySQLdb/connections.py", line 254, in query
    _mysql.connection.query(self, query)
MySQLdb.NotSupportedError: (1235, "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")
```

It's fine if you cast it to a list first though:

```python
>>> VLAN.objects.filter(vid__in=list(VLAN.objects.all().values_list("vid", flat=True)[:3]))
<RestrictedQuerySet [<VLAN: Layer (274)>, <VLAN: Detail (359)>, <VLAN: Wind (635)>]>
```